packages/next/build/webpack/plugins/next-trace-entrypoints-plugin.ts

```diff
@@ -32,6 +33,54 @@ function getModuleFromDependency(
   return compilation.moduleGraph.getModule(dep)
 }
 
+function getFilesMapFromReasons(
+  fileList: Set<string>,
+  reasons: NodeFileTraceReasons
+) {
+  // this uses the reasons tree to collect files specific to a
+  // certain parent allowing us to not have to trace each parent
+  // separately
+  const parentFilesMap = new Map<string, Set<string>>()
+
+  function propagateToParents(
+    parents: Set<string>,
+    file: string,
+    seen = new Set<string>()
+  ) {
+    for (const parent of parents || []) {
+      if (!seen.has(parent)) {
+        seen.add(parent)
+        let parentFiles = parentFilesMap.get(parent)
+
+        if (!parentFiles) {
+          parentFiles = new Set()
+          parentFilesMap.set(parent, parentFiles)
+        }
+        parentFiles.add(file)
+        const parentReason = reasons.get(parent)
+
+        if (parentReason?.parents) {
+          propagateToParents(parentReason.parents, file, seen)
+        }
+      }
+    }
+  }
+
+  for (const file of fileList!) {
+    const reason = reasons!.get(file)
```

Comment:
Should ! be ? Instead

Reply:
Oh looks like this was left over from a previous change as reasons should never be empty here. I wonder if there's a lint error for un-necessary non-null assertion 🤔